spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -1109,6 +1110,22 @@ public void testAddFilesPartitionedWithParallelism() {
         sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
   }
 
+  @TestTemplate
+  public void testMigrateWithInvalidParallelism() throws IOException {
+    createUnpartitionedHiveTable();
+
+    createIcebergTable(
+        "id Integer, name String, dept String, subdept String", "PARTITIONED BY (id)");
+
+    assertThatThrownBy(
+            () ->
+                sql(
+                    "CALL %s.system.add_files(table => '%s', source_table => '%s', parallelism => %d)",
+                    catalogName, tableName, sourceTableName, -1))
```

Comment:
nit: We can inline `-1`.

Reply:
Thank you for your suggestions. I have already made the changes to the code.